### DISCUSSION TITLE
How to remove/disable download button on code server?

### DISCUSSION THREAD

**Discussion by zhouzhou12** [coder/code-server#5886] [Q&A] [answered]
For data security,  my organization requires to prohibit downloading files from server. 
How to remove/disable the download button on the page of code server?
Who can help me?
Thanks!

**Comment by jsjoeio** [comment, marked as answer]
I was surprised we don't have notes in the docs on this. Apologies! Here are the docs for that:

```shell
      --disable-file-downloads           Disable file downloads from Code. This can also be set with CS_DISABLE_FILE_DOWNLOADS set to 'true' or '1'.
```

This is a flag you can pass to `code-server` on startup.

**Reply by zhouzhou12** [reply]
Thank you!

**Comment by rockan007** [comment]
How can i use it with the docker?

**Reply by code-asher** [reply]
You can add arguments to the end of the `docker run` command.  `docker run codercom/code-server --disable-file-downloads` for example.

**Comment by paynpro** [comment]
But its not working here i added in **sudo vi /etc/coder.d/coder.env** 

# e.g. https://coder.example.com
CODER_ACCESS_URL=http://13.21.177.12
CS_DISABLE_FILE_DOWNLOADS=true

and i run this command

**sudo systemctl daemon-reexec
sudo systemctl restart coder**

Now if i go for coder and open in **code-server** and i click any file and click **Download** option it is downloading . and also it is sending the request like
**blob:http://<ip or domain>/some keys**

so please tell me how to resolve this if i use code-server or vscode desktop if i click download it need to be give 403 error or it dont need to be download please give me the solution.

**Reply by code-asher** [reply]
I do not believe systemd will read that `coder.env` file, will it?  You probably have to set the environment variable in the service configuration file.  For example:

```
[Service]
Environment=CS_DISABLE_FILE_DOWNLOADS=true
```

**Reply by code-asher** [reply]
Except, I misread, you are starting Coder with systemd, not code-server, right?  I think environment variables set on coderd will not propagate to the workspace agent.

How are you starting code-server?  Are you using [the module](https://github.com/coder/registry/tree/main/registry/coder/modules/code-server)?  If so, I am not sure there is a way to configure this.  The module will have to be modified to support the flag.

But if you are not using the module and launching code-server yourself, you can just add the flag.  For example:

```
resource "coder_agent" "main" {
  arch           = data.coder_provisioner.me.arch
  os             = "linux"
  startup_script = <<-EOT
    set -e
    curl -fsSL https://code-server.dev/install.sh | sh -s -- --method=standalone --prefix=/tmp/code-server
    /tmp/code-server/bin/code-server --auth none --port 13337 --disable-file-downloads >/tmp/code-server.log 2>&1 &
  EOT
```

**Reply by code-asher** [reply]
Also note that this flag is for code-server only, it does not affect VS Code desktop.
